go/vt/vtgate/scatter_conn.go

```diff
@@ -676,15 +681,15 @@ func (stc *ScatterConn) ExecuteLock(
 
 	switch info.actionNeeded {
 	case nothing:
-		if reservedID == 0 {
-			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "[BUG] reserved id zero not expected %v", reservedID)
-		}
 		qr, err = qs.Execute(ctx, rs.Target, query.Sql, query.BindVariables, 0 /* transactionID */, reservedID, opts)
 		if err != nil && wasConnectionClosed(err) {
+			// TODO: try to acquire lock again.
```

Comment:
did you leave this here intentionally?

Reply:
yes, I believe we can try acquiring a lock here again, but not sure if this seems a good approach.